# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBTransaction.cpp
# Commit: a114b9b8e6ea
# Full Hash: a114b9b8e6eaa23a445a31efc436f4e367f7874f
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-01-10 21:49:10
# Regressor Bug: 1600906
# File Overlap Count: 1
# Description:
#   Bug 1600906 - Convert IDBCursor and BackgroundCursorChild to templates to increase type safety and reduce state. r=dom-workers-and-storage-reviewers,ytausky
#   
#   This also simplifies delegating calls that are dependent on the cursor type.
#   
#   Also reduce dependency on IDBCursor.h by moving enums and type traits to IDBCursorType.h
# ==============================================================================

diff -r cc3cfee4a014 -r a114b9b8e6ea dom/indexedDB/IDBTransaction.cpp
--- a/dom/indexedDB/IDBTransaction.cpp	Fri Jan 10 11:07:59 2020 +0000
+++ b/dom/indexedDB/IDBTransaction.cpp	Fri Jan 10 14:21:18 2020 +0000
@@ -299,8 +299,9 @@
   return actor;
 }
 
-void IDBTransaction::OpenCursor(BackgroundCursorChild* const aBackgroundActor,
-                                const OpenCursorParams& aParams) {
+void IDBTransaction::OpenCursor(
+    PBackgroundIDBCursorChild* const aBackgroundActor,
+    const OpenCursorParams& aParams) {
   AssertIsOnOwningThread();
   MOZ_ASSERT(aBackgroundActor);
   MOZ_ASSERT(aParams.type() != OpenCursorParams::T__None);